pkg/planner/core/operator/logicalop/logical_join.go

```diff
@@ -1080,6 +1085,46 @@ func (p *LogicalJoin) AppendJoinConds(eq []*expression.ScalarFunction, left, rig
 	p.OtherConditions = append(other, p.OtherConditions...)
 }
 
+func (p *LogicalJoin) isAllUniqueIDInTheSameTable(cond expression.Expression) bool {
+	colset := slices.Collect(
+		maps.Keys(expression.ExtractColumnsMapFromExpressions(nil, cond)),
+	)
+	if len(colset) == 1 {
+		return true
+	}
+	for _, schema := range p.allDataSouceSchema {
+		inTheSameSchema := true
+		for _, unique := range colset {
+			if !slices.ContainsFunc(schema.Columns, func(c *expression.Column) bool {
+				return c.UniqueID == unique
+			}) {
+				inTheSameSchema = false
+				break
+			}
+		}
+		if inTheSameSchema {
```

Comment:
There is also a quick path that if some(not all) colset belong to one schema, we can return false directly